### DISCUSSION TITLE
How can I use `updateQueryData` and stop the response from flowing into `selectFromResult` ?

### DISCUSSION THREAD

**Discussion by ghasemikasra39** [reduxjs/redux-toolkit#1559] [Q&A]
I noticed that when I use `selectFromResult` in my `UseQuery` hook, and manually update the cache using `updateQueryData`, the updated cache data will flow into `selectFromResult`. This behavior is surprising for me, and in my case it is undesired:

```javascript
  const {data} = useGetScrollingNoteQuery(
      queryInput, {
        selectFromResult: (queryResult) => { ... }
      })

// Later on

    dispatch(scrollingNoteApi.util.updateQueryData('getScrollingNote', queryInput,  (draftNotes) =>  { ...  } ))
```

Is there a way to stop this behavior? I.e. when I change the cache manually, I don't want the data to be pre-processed by `selectFromResult`, because I have not called `UseQuery` hook (it was no triggered).

**Comment by markerikson** [comment]
The hook simply reads the data that is cached in the store for that cache key.  When you use `updateQueryData`, you _are_ updating that value, and therefore the hook's selector will run.  When the hook runs, _you_ gave it `selectFromResult`, so it will use that.

What are you expecting to have happen instead?
